camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/mapred/EtlInputFormat.java

```diff
@@ -461,6 +464,17 @@ private void writeRequests(List<CamusRequest> requests, JobContext context) thro
     }
     return offsetKeysMap;
   }
+  
+  private PartitionMetadata refreshPartitionMetadata(PartitionMetadata partitionMetadata, TopicMetadata topicMetadata, JobContext context) {
+    List<TopicMetadata> topicMetadataList = this.getKafkaMetadata(context, Collections.singletonList(topicMetadata.topic()));
+    topicMetadata = topicMetadataList.get(0);
+    for(PartitionMetadata metadataPerPartition : topicMetadata.partitionsMetadata()) {
+      if (metadataPerPartition.partitionId() == partitionMetadata.partitionId()) {
+        return metadataPerPartition;
+      }
+    }
```

Comment:
If the loop finishes but no partition matches, it should be treated as a different error case.

And how did you test your code?

We also need to add unit test.